.github/workflows/build.yml

```diff
@@ -30,185 +34,295 @@ jobs:
 
-    - name: Patch GHC 8.10.2 linker
-      if: matrix.os == 'windows-latest' && matrix.ghc == '8.10.2'
-      run: |
-        sed -i \
-          's|C:/GitLabRunner/builds/2WeHDSFP/0/ghc/ghc/inplace/mingw/bin/ld.exe|C:/ProgramData/chocolatey/lib/ghc.8.10.2/tools/ghc-8.10.2/mingw/bin/ld.exe|g' \
-          C:/ProgramData/chocolatey/lib/ghc.8.10.2/tools/ghc-8.10.2/lib/settings
-
-    - name: Install build environment
-      if: matrix.os == 'ubuntu-latest'
+    - name: Show Haskell tool versions
       run: |
-        sudo apt-get update
-        sudo apt-get -y install libsodium23 libsodium-dev
-        sudo apt-get -y install libsystemd0 libsystemd-dev
-        sudo apt-get -y remove --purge software-properties-common
-        sudo apt-get -y autoremove
+        ghc --version
+        cabal --version
 
     - name: Set up temp directory
+      env:
+        RUNNER_TEMP: ${{ runner.temp }}
       run: |
-        echo 'TMPDIR=${{ runner.temp }}'  >> $GITHUB_ENV
-        echo 'TMP=${{ runner.temp }}'     >> $GITHUB_ENV
-
-    # Cache cabal-store
-    - name: cache cabal store
-      uses: actions/cache@v2
-      with:
-        path: ${{ steps.setup-haskell.outputs.cabal-store }}
-        key: cabal-store-${{ matrix.ghc }}-${{ matrix.os }}
-    # Cache parts of dist-newstyle (C:\dist)
-    - name: cache buiddir [cache]
-      uses: actions/cache@v2
-      with:
-        path: C:\dist\cache
-        key: dist-cache-${{ matrix.ghc }}-${{ matrix.os }}
-    - name: cache buiddir [src]
-      uses: actions/cache@v2
-      with:
-        path: C:\dist\src
-        key: dist-src-${{ matrix.ghc }}-${{ matrix.os }}
-    - name: cache builddir [packagedb]
-      uses: actions/cache@v2
-      with:
-        path: C:\dist\packagedb
-        key: dist-packagedb-${{ matrix.ghc }}-${{ matrix.os }}
+        echo "TMPDIR=$RUNNER_TEMP"  >> $GITHUB_ENV
+        echo "TMP=$RUNNER_TEMP"     >> $GITHUB_ENV
 
-    - name: update Hackage index
-      run: cabal v2-update
-
-    # checkout repository
-    - name: checkout ouroboros-network repository
-      uses: actions/checkout@v2
+    - name: Cabal update
+      run: cabal update
 
     - name: Cabal Configure
       run: cabal --builddir="$CABAL_BUILDDIR" configure --enable-tests
 
-    - name: Use cabal.project.local.windows
+    - name: Record dependencies
+      run: |
+        cat ${{ env.PLAN_JSON }} | jq -r '."install-plan"[].id' | sort | uniq > dependencies.txt
+
+    - uses: actions/cache@v2
+      name: Cache cabal store
+      with:
+        path: ${{ steps.setup-haskell.outputs.cabal-store }}
+        key: cache-${{ env.CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('dependencies.txt') }}
+        restore-keys: |
+          cache-${{ env.CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('dependencies.txt') }}
+          cache-${{ env.CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}-
+
+    - name: Use cabal.project.local.ci.windows
```

Comment:
Shouldn't we use this only on `Windows` as the name `cabal.project.local.ci.windows` suggests?

Reply:
Fixed.  Thanks!